### ISSUE TITLE
Resolving many Domain names at the same time can lead to infinite loop

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5048]
|<img alt="melug's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| melug reported|
|-|-|
|Trac ID|trac#5048|
|Type|defect|
|Created|2011-04-12 01:14:50Z|

Resolving many Domain names at the same time can lead to infinite loop. In twisted.names.client on line number 229:
```python
        proto = dns.DNSDatagramProtocol(self) 
        while True:
            try:
                self._reactor.listenUDP(dns.randomSource(), proto)
            except error.CannotListenError, e:
                pass           
            else:
                return proto
```
If CannotListenError was because of "[Errno 24] Too many open files." error then it will loop infinitely.

<details><summary>Searchable metadata</summary>

```
trac-id__5048 5048
type__defect defect
reporter__melug melug
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__duplicate duplicate
component__names names
keywords__ 
time__1302570890000000 1302570890000000
changetime__1555348895284102 1555348895284102
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

[#6145](https://github.com/twisted/twisted/issues/6145) was a duplicate of this.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_3"></a>|@hawkowl set status to `closed`|
|-|-|

Fixed in [#9620](https://github.com/twisted/twisted/issues/9620), closing this as a duplicate.
